src/Microsoft.AspNet.Mvc.Core/Infrastructure/ObjectResultExecutor.cs

```diff
@@ -0,0 +1,276 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Http;
+using Microsoft.AspNet.Mvc.Core;
+using Microsoft.AspNet.Mvc.Formatters;
+using Microsoft.AspNet.Mvc.Internal;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.OptionsModel;
+using Microsoft.Net.Http.Headers;
+
+namespace Microsoft.AspNet.Mvc.Infrastructure
+{
+    public class ObjectResultExecutor
+    {
+        public ObjectResultExecutor(
+            IOptions<MvcOptions> options,
+            IActionBindingContextAccessor bindingContextAccessor,
+            ILoggerFactory loggerFactory)
+        {
+            if (options == null)
```

Comment:
That's kinda arbitrary yeah? Happy to do whatever's standard here but there's nothing stopping someone from newing this up.